src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,134 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+
+public class RegistrationServiceImplTest {
+    private static final String PASSWORD_TEST = "123123";
+    private static final String LOGIN_TEST = "MyTestLogin";
+    private static final int AGE_TEST = 34;
+    private static RegistrationService service;
+
+    @BeforeAll
+    static void setUp() {
+        service = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void register_ValidUserSuccessfulRegistration_Ok() {
+        User validUser = new User(LOGIN_TEST, PASSWORD_TEST, AGE_TEST);
+        assertEquals(validUser, service.register(validUser));
+    }
+
+    @Test
+    void register_LoginNull_NotOk() {
+        User user = new User(null, PASSWORD_TEST, AGE_TEST);
+        assertThrows(RegistrationException.class, new Executable() {
+            @Override
+            public void execute() throws Throwable {
+                service.register(user);
+            }
+        });
+    }
+
+    @Test
+    void register_PasswordNull_NotOk() {
+        User user = new User(LOGIN_TEST, null, AGE_TEST);
+        assertThrows(RegistrationException.class, new Executable() {
+            @Override
+            public void execute() throws Throwable {
+                service.register(user);
+            }
+        });
+    }
+
+    @Test
+    void register_UserAlreadyExist_NotOk() {
+        User user = new User(LOGIN_TEST, PASSWORD_TEST, AGE_TEST);
+
+        Storage.PEOPLE.add(user);
+        assertThrows(RegistrationException.class,
```

Comment:
Better to create another use with the same login. 
```suggestion
        User newUser = new User(...parameters....);
        assertThrows(RegistrationException.class, () -> registrationService.register(newUser));
```